### DISCUSSION TITLE
Can't find variable globalThis

### DISCUSSION THREAD

**Discussion by feargalWag** [pmndrs/jotai#2091] [Q&A]
Shortly after pushing Jotai to production we have started getting errors on pages that _don't_ reference Jotai APIs referring to `globalThis`. We have tried 
1. Adding a polyfill for `globalThis` to the root app.tsx file of our NextJS app.
2. Moving the `<Provider>` component from Jotai to only wrap root components of pages where Jotai is used.
3. Removed the `jotai/babel/plugin-react-refresh` plugin. 

Jotai references globalThis heavily in our sourcemap.

**Comment by dai-shi** [comment]
If removing `jotai/babel/plugin-react-refresh` doesn't solve it, only the remaining is the one in `store.ts`, but that should only be effective in non production mode.
We can add a guard, but it sounds like the production build doesn't work as expected.

What exactly is the error message for `globalThis`?
